src/textual/app.py

```diff
@@ -206,6 +208,37 @@ def isatty(self) -> bool:
         return True
 
 
+class _PrintCapture:
+    """A file-like which captures output."""
+
+    def __init__(self, app: App, stderr: bool = False) -> None:
+        """
+
+        Args:
+            app: App instance.
+            stderr: Write from stderr.
+        """
+        self.app = app
+        self.stderr = stderr
+
+    def write(self, text: str) -> None:
+        """Captured write.
+
+        Args:
+            text: Text that was "printed".
+        """
+        self.app._print(text, stderr=self.stderr)
+
+    def flush(self) -> None:
+        """Stdout or stderr was flushed."""
```

Comment:
This docstring reads more like one for a boolean property than one for a method that performs an action. Might want to tweak it it?